numpy/lib/function_base.py

```diff
@@ -764,6 +764,22 @@ def copy(a, order='K', subok=False):
     >>> x[0] == z[0]
     False
 
+    Note that ``np.copy`` is a shallow copy and will not
+    copy object elements within arrays:
+
+    >>> obj = dict()
+    >>> a = np.array([obj], dtype=object)
+    >>> b = np.copy(a)
+    >>> b[0] is a[0]
+    True
+
+    To copy the elements as well, use `copy.deepcopy`:
+
+    >>> import copy
+    >>> c = copy.deepcopy(a)
+    >>> c[0] is a[0]
+    False
```

Comment:
```suggestion
    >>> import copy
    >>> a = np.array([1, 'm', [2, 3, 4]], dtype=object)
    >>> c = copy.deepcopy(a)
    >>> c[2][0] = 10
    >>> c
    array([1, 'm', list([10, 3, 4])], dtype=object)
    >>> a
    array([1, 'm', list([2, 3, 4])], dtype=object)
```
Extension of the above suggested example if you decide to go with it.